discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -167,22 +167,23 @@ def generate_and_download_infra_env(
         ssh_key=None,
         ignition_info=None,
         proxy=None,
-    ):
+    ) -> Path:
+        if self._config.is_static_ip and static_network_config is None:
```

Comment:
What does this have to do with changing how we download the image?

Reply:
I moved that from `download_image` to here.
It's needed for generating the `infra_env`